### DISCUSSION TITLE
Import of procedural materials from blender to Godot engine for .glb and .blend formats

### DISCUSSION THREAD

**Discussion by chloestudio** [godotengine/godot-proposals#8929] [3D]
I suggest that a better import of the materials that are created procedurally in Blender into Godot engine should be implemented, since if I have an object that I exported from Blender to Godot with procedural textures they do not work and the colors end up looking completely flat.
![prueba](https://github.com/godotengine/godot-proposals/assets/141695032/621ee945-9647-4585-be69-4e0f20b1cb32)
![prueba_2](https://github.com/godotengine/godot-proposals/assets/141695032/5edbefc8-194b-4281-aa16-6f81ecf25244)
![prueba_3](https://github.com/godotengine/godot-proposals/assets/141695032/26e21328-2088-4378-af92-9948658273ca)

**Comment by AThousandShips** [comment]
These material systems aren't supported, you need to bake them or otherwise convert them, is this part of the glb specification or specifically Blender? Does other software support importing them like this?

**Reply by chloestudio** [reply]
blender packages the object materials in glb format, but what are procedural textures as materials do not work, I know that the main process is to bake the textures, but I was experimenting if exporting the .blend format would recognize my procedural materials created in blender. It would be great if these materials could be exported from blender to godot

**Reply by AThousandShips** [reply]
Probably not supported, I think they need to be baked first, this is a very large and complex feature that should be handled on the export end, which I believe is possible, it'd be a lot of work to keep up with these features and better to not do that if it's possible to bake it

**Reply by chloestudio** [reply]
I'm going to bake the procedural textures from blender to avoid having these inconveniences, I was only doing it as a test to see if these materials could be exported with their procedural nodes to godot engine, that's why I suggested if this connection could be made with the formats .blend or that coupling of the two programs to facilitate the workflow, but it will still have to be done the traditional way

**Reply by AThousandShips** [reply]
Related:
* https://github.com/godotengine/godot-proposals/issues/714
